presto-main/src/main/java/com/facebook/presto/type/MapType.java

```diff
@@ -69,6 +72,115 @@ public Type getValueType()
     }
 
     @Override
+    public boolean isComparable()
+    {
+        return valueType.isComparable();
+    }
+
+    @Override
+    public int hash(Block block, int position)
+    {
+        Slice mapSlice = getSlice(block, position);
+        Block mapBlock = readStructuralBlock(mapSlice);
+        Map<Integer, Integer> hashMap = new HashMap<>();
+
+        for (int i = 0; i < mapBlock.getPositionCount(); i += 2) {
+            //TODO fix collision, hashPosition could always return 0
+            hashMap.put(hashPosition(keyType, mapBlock, position), hashPosition(valueType, mapBlock, position + 1));
```

Comment:
We need to fix the issue with hash collisions. I think the best way to do that is to calculate the hash code as the sum of all the hashes of the keys and values